src/lib/consensus/proof_of_stake.ml

```diff
@@ -2081,6 +2081,9 @@ module Data = struct
       ; curr_epoch= Segment_id.to_int t.curr_epoch
       ; curr_slot= Segment_id.to_int t.curr_slot
       ; total_currency= Amount.to_int t.total_currency }
+
```

Comment:
Why does this have to be an option. It looks like the value is always a `Some` type.

Reply:
There could be a consensus mechanism that doesn't have notions of epoch and slot. That was true for posig, for example.

An alternative would be for that other mechanism to raise an exception. But then we'd have to handle that exception.